### PR TITLE
feat(navigation): replace TanStack Router with nanostore CSS show/hide

### PR DESCRIPTION
**Pull request by DaniAkash**
## Summary

- Replaces TanStack Router with a `$navigation.ts` nanostore (`$activeProjectId`, `$activeTabId`) — no more URL-driven routing in a native desktop app
- All terminal instances stay mounted permanently; tab and project switches are instant CSS `display` toggles, preserving ghostty-web canvas state, pty processes, and scrollback history
- New `WorkspaceView.tsx` lazy-mounts each `TerminalPane` on first visit then CSS-hides/shows on subsequent visits
- Full 22-case `bun test` suite for `$navigation.ts` covering init, project nav, tab nav, and tab removal edge cases
- Removes `@tanstack/react-router`, `@tanstack/router-devtools`, `@tanstack/router-plugin`, all route files

## What changed

| File | Change |
|------|--------|
| `src/modules/stores/$navigation.ts` | New — navigation atoms + 4 pure functions |
| `src/modules/stores/$navigation.test.ts` | New — 22 bun tests, all passing |
| `src/screens/workspace/WorkspaceView.tsx` | New — CSS show/hide tab panes per project |
| `src/screens/workspace/WorkspaceLayout.tsx` | Rewrite — renders all projects, module-level keyboard handler |
| `src/components/Sidebar/SidebarProjectRow.tsx` | `useParams` → `useStore($activeProjectId)` |
| `src/components/Sidebar/SidebarTabItem.tsx` | `useNavigate/useParams` → `navigateToTab` |
| `src/components/TabBar/TabBar.tsx` | `useNavigate/useParams` → `navigateToTab/onTabRemoved` |
| `src/components/StatusBar/StatusBar.tsx` | Self-contained via `$projects` store |
| `src/main.tsx` | `listProjects → $projects → initNavigation → render` |
| `vite.config.ts` | Remove `TanStackRouterVite` plugin |
| `biome.json` | Remove `routeTree.gen.ts` exclusion |
| `tsconfig.json` | Exclude `*.test.ts` from tsc (bun handles test types) |
| deleted | All `src/routes/**`, `src/router.tsx`, `src/routeTree.gen.ts`, old terminal screens |

## Test plan

- [ ] `bun test` — 22 tests pass (run in worktree)
- [ ] `bun run typecheck` — no errors
- [ ] `bun run lint` — clean
- [ ] App starts: first project and first tab auto-selected via `initNavigation`
- [ ] Click a project row in sidebar → workspace switches to that project
- [ ] Click a tab in sidebar or tab bar → CSS show/hide, no terminal remount
- [ ] Switch to a tab never visited → TerminalPane mounts, shell starts up
- [ ] Return to a previously visited tab → history preserved, no remount
- [ ] Switch projects → return to previous project → history preserved
- [ ] Cmd+T adds tab, Cmd+W closes active tab, Cmd+1–9 jumps to tab
- [ ] Pin/unpin and drag-to-reorder still work for both projects and tabs

### CODE REVIEW

File: src/screens/workspace/WorkspaceLayout.tsx [hunk 1]
```diff
@@ -1,107 +1,88 @@
...
+            <div
+              key={project.id}
+              className="absolute inset-0 flex flex-col"
+              style={{
+                display: project.id === activeProjectId ? 'flex' : 'none',
+              }}
+            >
+              <WorkspaceView project={project} />
+            </div>
+          ))}
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src/screens/workspace/WorkspaceLayout.tsx]: Fixed — added a `mountedProjects` Set in WorkspaceLayout that lazy-mounts WorkspaceView only on first project activation, same pattern as the per-tab `mounted` Set in WorkspaceView.

File: src/screens/workspace/WorkspaceView.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+export function WorkspaceView({ project }: Props) {
+  const activeTabsByProject = useStore($activeTabId)
+  const activeTabId =
+    activeTabsByProject[project.id] ?? project.tabs[0]?.id ?? ''
+
+  // Lazy mount: only create a TerminalPane when a tab is first visited.
+  // Once mounted, the pane stays alive forever (CSS show/hide, not unmount).
+  const [mounted, setMounted] = useState<Set<string>>(
+    () => new Set(activeTabId ? [activeTabId] : []),
+  )
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src/screens/workspace/WorkspaceView.tsx:22]: Fixed — removed the `?? project.tabs[0]?.id` fallback; `activeTabId` now uses only the store value (`?? ''`), so `mounted` starts empty for unvisited projects and is populated only by `initNavigation`/`navigateToProject`.

File: src/screens/workspace/WorkspaceLayout.tsx [hunk 3]
```diff
@@ -1,107 +1,88 @@
...
-    })
+  if (!project) return
+  const tabId = $activeTabId.get()[projectId] ?? ''
+  if (e.key === 't') {
+    e.preventDefault()
+    handleNewTab(projectId)
+  } else if (e.key === 'w') {
+    e.preventDefault()
+    handleCloseTab(projectId, tabId)
+  } else if (/^[1-9]$/.test(e.key)) {
```
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src/screens/workspace/WorkspaceLayout.tsx:50]: Fixed — added `if (!tabId) return` guard in the Cmd+W branch before calling `handleCloseTab`.

File: src/screens/workspace/WorkspaceLayout.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (DaniAkash) [reply src/screens/workspace/WorkspaceLayout.tsx]: Fixed — added a `useEffect` watching `[projects, activeProjectId]` that calls `navigateToProject(projects[0].id)` whenever the active project no longer exists in the list.
